src/EditorFeatures/Core/InheritanceMargin/AbstractInheritanceMarginService.cs

```diff
@@ -0,0 +1,159 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.FindSymbols.FindReferences;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.SymbolMapping;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.InheritanceMargin
+{
+    internal abstract partial class AbstractInheritanceMarginService : IInheritanceMarginService
+    {
+        public async Task<ImmutableArray<InheritanceMemberItem>> GetInheritanceInfoAsync(
+            Document document,
+            CancellationToken cancellationToken)
+        {
+            var options = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+            var featureEnabled = options.GetOption(InheritanceMarginOptions.ShowInheritanceMargin);
```

Comment:
should this not be in the view tagger instead?  the service shouldn't care.

Reply:
Moved